--- a/lib/src/localizations/localizations_utils.dart
+++ b/lib/src/localizations/localizations_utils.dart
@@ -8,8 +8,6 @@
 import '../base/common.dart';
 import '../base/file_system.dart';
 import '../base/logger.dart';
-import '../globals.dart' as globals;
-import '../runner/flutter_command.dart';
 import 'gen_l10n_types.dart';
 import 'language_subtag_registry.dart';
 
@@ -534,47 +532,6 @@
   );
 }
 
-/// Parse the localizations configuration from [FlutterCommand].
-LocalizationOptions parseLocalizationsOptionsFromCommand({
-  required FlutterCommand command,
-  required String defaultArbDir,
-}) {
-  const kSyntheticPackage = 'synthetic-package';
-  const kFlutterGenNotice = 'http://flutter.dev/to/flutter-gen-deprecation';
-  if (command.argResults!.wasParsed(kSyntheticPackage)) {
-    if (command.boolArg(kSyntheticPackage)) {
-      throwToolExit(
-        'Cannot enable "$kSyntheticPackage", this feature has been removed. '
-        'See $kFlutterGenNotice.',
-      );
-    } else {
-      globals.logger.printWarning(
-        'The argument "$kSyntheticPackage" no longer has any effect and should '
-        'be removed. See $kFlutterGenNotice',
-      );
-    }
-  }
-  return LocalizationOptions(
-    arbDir: command.stringArg('arb-dir') ?? defaultArbDir,
-    outputDir: command.stringArg('output-dir'),
-    outputLocalizationFile: command.stringArg('output-localization-file'),
-    templateArbFile: command.stringArg('template-arb-file'),
-    untranslatedMessagesFile: command.stringArg('untranslated-messages-file'),
-    outputClass: command.stringArg('output-class'),
-    header: command.stringArg('header'),
-    headerFile: command.stringArg('header-file'),
-    useDeferredLoading: command.boolArg('use-deferred-loading'),
-    genInputsAndOutputsList: command.stringArg('gen-inputs-and-outputs-list'),
-    projectDir: command.stringArg('project-dir'),
-    requiredResourceAttributes: command.boolArg('required-resource-attributes'),
-    nullableGetter: command.boolArg('nullable-getter'),
-    format: command.boolArg('format'),
-    useEscaping: command.boolArg('use-escaping'),
-    suppressWarnings: command.boolArg('suppress-warnings'),
-    useNamedParameters: command.boolArg('use-named-parameters'),
-  );
-}
-
 // Try to read a `bool` value or null from `yamlMap`, otherwise throw.
 bool? _tryReadBool(YamlMap yamlMap, String key, Logger logger) {
   final Object? value = yamlMap[key];
