--- a/lib/src/base/process.dart
+++ b/lib/src/base/process.dart
@@ -8,9 +8,7 @@
 import 'package:process/process.dart';
 
 import '../convert.dart';
-import '../globals.dart' as globals;
 import 'async_guard.dart';
-import 'exit.dart';
 import 'io.dart';
 import 'logger.dart';
 import 'utils.dart';
@@ -667,36 +665,3 @@
     }
   }
 }
-
-Future<int> exitWithHooks(int code, {required ShutdownHooks shutdownHooks}) async {
-  if (globals.analytics.shouldShowMessage) {
-    globals.logger.printStatus(globals.analytics.getConsentMessage);
-    globals.analytics.clientShowedMessage();
-
-    // This trace is searched for in tests.
-    globals.logger.printTrace('Showed analytics consent message.');
-  }
-
-  // Run shutdown hooks before flushing logs
-  await shutdownHooks.runShutdownHooks(globals.logger);
-
-  final completer = Completer<void>();
-
-  await globals.analytics.close();
-
-  // Give the task / timer queue one cycle through before we hard exit.
-  Timer.run(() {
-    try {
-      globals.printTrace('exiting with code $code');
-      exit(code);
-      completer.complete();
-      // This catches all exceptions because the error is propagated on the
-      // completer.
-    } catch (error, stackTrace) {
-      completer.completeError(error, stackTrace);
-    }
-  });
-
-  await completer.future;
-  return code;
-}
